kofta/src/app/utils/getBadWords.ts

```diff
@@ -0,0 +1,15 @@
+import fs from 'fs'
+
+// This will get the bad words from the txt file and return the array
+export const getBadWords = (): Array<string> => {
+    let returnArr: string = ''
+
+    fs.readFile('../../list_of_bad_words.txt', 'utf8', function (err,data) {
+        if (err) {
+          return console.log(err);
+        }
+        returnArr = data
```

Comment:
`fs.readFile` is asynchronous. The callback function executes *after* the enclosing function returns, so `returnArr` will still be empty when `getBadWords` returns.

That said, this is a front-end web app; `fs` is a Node.js module. Instead of fs, just store an array of all words in a json file and import it that way.

Reply:
ok

Reply:
tho i think fs would still work.

Reply:
`fs` does not work on frontend, there is no filesystem.

Reply:
Also @yummyweb looking at how dogehouse is structured i think all of the logic should be on the back end, people can circumvent this easily

Reply:
I agree